plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -689,7 +689,9 @@ private void setColumnStatistics(String objectName, List<ColumnStatisticsObj> st
         verify(!dateStatistics.isEmpty() && metastoreSupportsDateStatistics.equals(Optional.empty()));
 
         try (ThriftMetastoreClient client = createMetastoreClient()) {
-            saveColumnStatistics.call(client, statisticsExceptDate);
+            if (!statisticsExceptDate.isEmpty()) {
```

Comment:
Could you add a product test? I confirmed we can reproduce the issue in `EnvSinglenode`.